docs/intro/quickstart/working.rst

```diff
@@ -0,0 +1,385 @@
+    +               order: card.order,
+    +             })
+    +           ),
+    +         })
+    +     )
+    +     .run(client, deck);
+
+        revalidatePath("/");
+      }
+
+.. edb:split-section::
+
+  Next, you will update the Server Actions for each ``Deck`` object: ``updateDeck``, ``addCard``, and ``deleteCard``. Start with ``updateDeck``, which is the most complex because it is dynamic. You can set either the ``title`` or ``description`` fields in an update. Use the dynamic nature of the query builder to generate separate queries based on which fields are present in the form data.
+
+  This may seem a bit intimidating at first, but the key to making this query dynamic is the ``nameSet`` and ``descriptionSet`` variables. These variables conditionally add the ``name`` or ``description`` fields to the ``set`` parameter of the ``update`` call.
+
+  .. code-block:: typescript-diff
+    :caption: app/(authenticated)/deck/[id]/actions.ts
+
+      "use server";
+
+      import { revalidatePath } from "next/cache";
+    - import { readFile, writeFile } from "node:fs/promises";
+    + import { client } from "@/lib/gel";
+    + import e from "@/dbschema/edgeql-js";
+      import { Deck } from "@/lib/models";
+
+      export async function updateDeck(formData: FormData) {
+        const id = formData.get("id");
+        const name = formData.get("name");
+        const description = formData.get("description");
+
+        if (
+          typeof id !== "string" ||
+          (typeof name !== "string" &&
+          typeof description !== "string")
+        ) {
+          return;
+        }
+
+    -   const decks = JSON.parse(
+    -     await readFile("./decks.json", "utf-8")
+    -   ) as Deck[];
+    -   decks[index].name = name ?? decks[index].name;
+    +   const nameSet = typeof name === "string" ? { name } : {};
+    -   decks[index].description = description ?? decks[index].description;
+    +   const descriptionSet =
+    +     typeof description === "string" ? { description: description || null } : {};
+
+    +   await e
+    +     .update(e.Deck, (d) => ({
+    +       filter_single: e.op(d.id, "=", id),
+    +       set: {
+    +         ...nameSet,
+    +         ...descriptionSet,
+    +       },
+    +     })).run(client);
+    -   await writeFile("./decks.json", JSON.stringify(decks, null, 2));
+        revalidatePath(`/deck/${id}`);
+      }
+
+    + const addCardQuery = e.params(
+    +   {
+    +     front: e.str,
+    +     back: e.str,
+    +     deckId: e.uuid,
+    +   },
+    +   (params) => {
+    +     const deck = e.assert_exists(
+    +       e.select(e.Deck, (d) => ({
+    +         filter_single: e.op(d.id, "=", params.deckId),
+    +       }))
+    +     );
+    +
+    +     const order = e.cast(e.int64, e.max(deck.cards.order));
+    +     return e.insert(e.Card, {
+    +       front: params.front,
+    +       back: params.back,
+    +       deck: e.cast(e.Deck, params.deckId),
```

Comment:
is this supposed to work? deck is not part of the Card type

Reply:
yeah, Aleksandra also caught that. We used to have a slightly different data model, but I need to update this one.